### PR TITLE
Changelog updates since v4.1.1

### PR DESCRIPTION
**Pull request by milesfrain**
**Description of the change**

This PR updates the CHANGELOG in preparation for the upcoming release of this library.

Related: purescript/purescript#3985

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,12 +5,22 @@ Notable changes to this project are documented in this file. The format is based
 ## [Unreleased]
 
 Breaking changes:
+- Updated code for PureScript 0.14 (#120)
+- Removed `fold1Default` and deprecated `foldMap1Default` (#128)
```
Comment (milesfrain) [inline CHANGELOG.md:9]: I believe this was the net result of breaking changes. Would be good to double-check.

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,12 +5,22 @@ Notable changes to this project are documented in this file. The format is based
...
 
 New features:
+- Added `findMapWithIndex` (#119)
+- Added `foldr1` `foldl1` `foldr1Default` `foldl1Default` `foldMap1DefaultR` `foldMap1DefaultL` (#121, #128)
```
Comment (milesfrain) [inline CHANGELOG.md:13]: Ignoring the introduction, then removal of `fold1DefaultR` `fold1DefaultL`

Comment (thomashoneyman) [inline CHANGELOG.md:13]: Should these be comma-separated, ie. 

```
- Added `foldr1`, `foldl1`, ..., and `foldMap1DefaultL`
```

File: CHANGELOG.md (hunk 1 above)
Comment (thomashoneyman) [reply CHANGELOG.md:9]: I think this is correct

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,12 +5,22 @@ Notable changes to this project are documented in this file. The format is based
 ## [Unreleased]
 
 Breaking changes:
+- Updated code for PureScript 0.14 (#120)
```
Comment (hdgarrood) [inline CHANGELOG.md:8]: I think it's clearer to phrase this as "Added support for PureScript 0.14, and dropped support for all previous versions"

**Review by thomashoneyman — APPROVED** [review]

**Pull request merged**